manifests/config.pp

```diff
@@ -47,7 +47,7 @@
     }
   }
 
-  if $::vault::install_method == 'archive' {
+  if $::vault::install_method == 'archive' or $::vault::manage_service_file {
```

Comment:
@jsok I removed the $::vault::install_method  and just used the $::vault::manage_service_file  condition to manage the service script. Fits all cases.